libs/src/services/discoveryProvider/index.js

```diff
@@ -28,14 +27,25 @@ class DiscoveryProvider {
 
   async init () {
     let endpoint
+    let pick
+
     if (this.autoselect) {
       endpoint = await this.autoSelectEndpoint()
     } else {
       if (!this.whitelist || this.whitelist.size === 0) {
         throw new Error('Must pass autoselect true or provide whitelist.')
       }
 
-      const pick = _.sample([...this.whitelist]) // selects random element from list.
+      // use this as a lookup between health_check endpoint and base url
+      const whitelistMap = {}
+      ;[...this.whitelist].map(url => {
+        whitelistMap[urlJoin(url, '/health_check')] = url
+      })
+
+      await Utils.raceRequests(Object.keys(whitelistMap), (url) => {
+        pick = whitelistMap[url]
+      }, {})
+
       const isValid = await this.ethContracts.validateDiscoveryProvider(pick)
```

Comment:
@raymondjacobson updated to use raceRequests. I noticed that with this current structure we make two requests to the successful discprov on init since we call validateDiscoveryProvider as well. I can try to remove the second request but what do you think of this direction?

Reply:
100%, let's get rid of it